internal/identify/studio.go

```diff
@@ -2,51 +2,128 @@ package identify
 
 import (
 	"context"
-	"database/sql"
-	"fmt"
-	"time"
+	"strconv"
 
-	"github.com/stashapp/stash/pkg/hash/md5"
 	"github.com/stashapp/stash/pkg/models"
+	"github.com/stashapp/stash/pkg/utils"
 )
 
 type StudioCreator interface {
-	Create(ctx context.Context, newStudio models.Studio) (*models.Studio, error)
-	UpdateStashIDs(ctx context.Context, studioID int, stashIDs []models.StashID) error
+	Create(ctx context.Context, input models.StudioDBInput) (*int, error)
 }
 
 func createMissingStudio(ctx context.Context, endpoint string, w StudioCreator, studio *models.ScrapedStudio) (*int, error) {
-	created, err := w.Create(ctx, scrapedToStudioInput(studio))
+	var dbInput models.StudioDBInput
+	var err error
+
+	if studio.Parent != nil {
+		if studio.Parent.StoredID == nil {
+			// The parent needs to be created
+			dbInput.ParentCreate, err = studioFromScrapedStudio(ctx, studio.Parent, endpoint)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			// The parent studio matched an existing one and the user has chosen in the UI to link and/or update it
+			dbInput.ParentUpdate, err = studioPartialFromScrapedStudio(ctx, studio.Parent, studio.Parent.StoredID, endpoint)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	dbInput.StudioCreate, err = studioFromScrapedStudio(ctx, studio, endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	studioID, err := w.Create(ctx, dbInput)
 	if err != nil {
-		return nil, fmt.Errorf("error creating studio: %w", err)
+		return nil, err
 	}
 
-	if endpoint != "" && studio.RemoteSiteID != nil {
-		if err := w.UpdateStashIDs(ctx, created.ID, []models.StashID{
+	return studioID, nil
+}
+
+// Duplicated in task_stash_box_tag.go
+func studioFromScrapedStudio(ctx context.Context, input *models.ScrapedStudio, endpoint string) (*models.Studio, error) {
+	// Populate a new studio from the input
```

Comment:
I wasn't able to find a good way to make this method and the next reusable in both places, hence the duplication.